src/Middleware/WsseMiddleware.php

```diff
@@ -2,75 +2,30 @@
 
 namespace CodeDredd\Soap\Middleware;
 
+use Http\Client\Common\Plugin;
 use Http\Promise\Promise;
-use Phpro\SoapClient\Middleware\Middleware;
-use Phpro\SoapClient\Xml\SoapXml;
 use Psr\Http\Message\RequestInterface;
 use Psr\Http\Message\ResponseInterface;
 use RobRichards\WsePhp\WSSESoap;
 use RobRichards\XMLSecLibs\XMLSecurityKey;
+use Soap\Psr18Transport\Xml\XmlMessageManipulator;
+use VeeWee\Xml\Dom\Document;
 
-class WsseMiddleware extends Middleware
+class WsseMiddleware implements Plugin
```

Comment:
Any specific reason why you don't use the plugin from php-soap?
Maybe we could improve that one so that it fits all needs?

Reply:
@veewee yes....i had the situation where a wsse header is needed without ssl permission keys. If you look at my middleware you see that.
Another thing was an Issue from someone where he needed wsse without `mustunderstand` which by default is always on in your middleware.
If you like i can make an PR for that so i can remove my middleware 😄

Reply:
Sure, feel free to do so. Seems right to include those cases.

FYI In the long run, it would be nice to make WSSE easier to configure.
Was thinking about creating something that works similar to SOAPUI for configuring it - in steps.
https://www.soapui.org/docs/soapui-projects/ws-security/

Reply:
Yeah your idea sound right. Mostly i try to have to configuration apporaches which are the most used.
One by configuration array and the other by function concatination. The second one you already do very well.